File: Algorithms/Java/1_Two_Sum/twoSum.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        HashMap<Integer, Integer> checkTable = new HashMap<Integer, Integer>();
        int[] result = {-1, -1};
        
        for (int i = 0; i < numbers.length; i++) {
            if (!checkTable.containsKey(new Integer(target - numbers[i]))) {
                checkTable.put(new Integer(numbers[i]), new Integer(i));
            } else {
                result[0] = checkTable.get(target - numbers[i]).intValue() + 1;
                result[1] = i + 1;
            }
        }
        return result;
    }
}

File: Algorithms/Python/1_Two_Sum/twoSum.py
class Solution:
    # @return a tuple, (index1, index2)
    def twoSum(self, num, target):
        checkTable = {}
        for i in xrange(0, len(num)):
            if checkTable.has_key(target - num[i]):
                return (checkTable.get(target - num[i]) + 1, i + 1)
            else:
                checkTable[num[i]] = i